### PR TITLE
Added pull request template

### LINKED ISSUE #851: Add PR template to improve GitHub Community Standards
The [GitHub Community Standards](https://github.com/SteeltoeOSS/Steeltoe/community) for this repo have all the boxes checked, except for a PR template.

Examples from other projects:
- https://github.com/json-api-dotnet/JsonApiDotNetCore/blob/master/.github/PULL_REQUEST_TEMPLATE.md
- https://github.com/dotnet/aspnetcore/blob/main/.github/pull_request_template.md
- https://github.com/dotnet/efcore/blob/main/.github/PULL_REQUEST_TEMPLATE.md
- https://github.com/fluentassertions/fluentassertions/blob/develop/.github/PULL_REQUEST_TEMPLATE.md
- https://github.com/xunit/xunit/blob/main/.github/PULL_REQUEST_TEMPLATE.md
- https://github.com/Humanizr/Humanizer/blob/main/.github/PULL_REQUEST_TEMPLATE.md
- https://github.com/App-vNext/Polly/blob/master/.github/pull_request_template.md
- https://github.com/icsharpcode/SharpZipLib/blob/master/.github/PULL_REQUEST_TEMPLATE.md

There's some guidance at https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/ and https://www.gitkraken.com/blog/enhancing-pull-request-descriptions-templates, but I think we should keep it as basic as possible. The longer it gets, the more people are going to dislike it (so delete it all upfront or refrain from opening PRs). I've seen that happen with the issue template at a former open-source project I maintained: once we required more info, people started using the chat instead.

In my opinion, the PR template should just be a basic checklist like https://github.com/json-api-dotnet/JsonApiDotNetCore/blob/master/.github/PULL_REQUEST_TEMPLATE.md. Such a quick reminder helped me on several occasions when I forgot to update the docs.

Feedback welcome!

### PR DESCRIPTION
**Pull request by bart-vmware**
Fixes #851.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<!-- Thank you for submitting a pull request to our repo. -->
+
+<!-- If this is your first PR in this repo, please read our [Contributing Guidelines (https://github.com/SteeltoeOSS/.github/blob/master/CONTRIBUTING.md) and remember to sign the [Contributor License Agreement](https://github.com/SteeltoeOSS/.github/blob/main/contributing-docs/contributing-license.md). Our bot will notify you shortly after the PR has been created. -->
+
+## Summary
+
+{Summary of the changes, less than 80 chars}
+
+## Description
```
Comment (TimHess) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion
## Description <!-- optional, no need to duplicate the summary -->
```

**Review by TimHess — COMMENTED** [review]
This generally works for me, I just want to make sure cases like first-timers making simple changes aren't intimidated by needing Summary + Description

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (bart-vmware) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: How about we drop the Summary/Description and replace it with just Description? The "less than 80 chars" feels a bit patronizing anyway.

Comment (TimHess) [reply .github/PULL_REQUEST_TEMPLATE.md:9]: Yes, that works

**Review by hananiel — DISMISSED** [review]
LGTM

**Review by dtillman — APPROVED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<!-- Thank you for submitting a pull request to our repo. -->
```
Comment (ccheetham) [inline .github/PULL_REQUEST_TEMPLATE.md:1]:
```suggestion
<!-- Thank you for submitting a pull request! -->
```

**Review by ccheetham — COMMENTED** [review]
one suggestion, otherwise looks good!

**Review by ccheetham — DISMISSED** [review]
lgtm

**Review by TimHess — APPROVED** [review]

**Pull request merged**